### DISCUSSION TITLE
what's the problem means? and how can i solve it ?

### DISCUSSION THREAD

**Discussion by Geotechnique-simlation** [Anura3D/Anura3D_OpenSource#28] [Q&A] [answered]
Error: time step : 5099, maximum number of recursions reached  trying to write results

**Comment by murphy-john** [comment, marked as answer]
This error message occurs in only one place in the code. It means one of your material points left it's old element and a new element could not be found for that material point. Have you tried using the debug feature to determine which material point it is?

**Reply by Geotechnique-simlation** [reply]
Thank you for your help. How can I debug the problem? I have seen the prior problem delivered by the other scholar， the debug method is a common way to solve the problems, but there is no an information to tell us how to use.

**Reply by murphy-john** [reply]
Debugging is a feature in Visual Studio. You can learn about how to debug [here](https://docs.microsoft.com/en-us/visualstudio/debugger/debugging-absolute-beginners?view=vs-2022&tabs=csharp). The link talks about another programming language but the same idea applies to Fortran.

**Comment by Geotechnique-simlation** [comment]
what is the difference between the functions of initial veloctiy and prescribed velocity?

**Reply by ayerrocolom** [reply]
Initial velocity is only applied in the first time step; then, the kinematics evolve based on the stability of the system.
Prescribed velocity is applied during the full simulation.
